scripts/source/livecompare.js

```diff
@@ -59,29 +65,40 @@ const destination = path.resolve(args[1]);
     for (const navTitle in dirEntry) {
       const fileAbsolutePath = path.resolve(basePath, dirEntry[navTitle]);
       const filename = path.relative(basePath, fileAbsolutePath);
-      const destFilepath = path.resolve(destPath, filename.replace(/\//g, '_')+"x");
-  
-      fileToMetadata[filename] = Object.assign({}, fileToMetadata[filename], {navTitle});
-      fileToMetadata[indexFilename].navigation = fileToMetadata[indexFilename].navigation||[];
-      fileToMetadata[indexFilename].navigation.push(path.basename(destFilepath, ".mdx"));
+      const destFilepath = path.resolve(
+        destPath,
+        filename.replace(/\//g, "_") + "x",
+      );
+
+      fileToMetadata[filename] = Object.assign({}, fileToMetadata[filename], {
+        navTitle,
+      });
+      fileToMetadata[indexFilename].navigation =
+        fileToMetadata[indexFilename].navigation || [];
+      fileToMetadata[indexFilename].navigation.push(
+        path.basename(destFilepath, ".mdx"),
+      );
 
       if (filename === indexFilename) continue;
       process(fileAbsolutePath, filename, destFilepath);
     }
   }
 
   // write out index w/ navigation tree
-  process(path.resolve(basePath, indexFilename), indexFilename, path.resolve(destPath, indexFilename+"x"));
+  process(
+    path.resolve(basePath, indexFilename),
+    indexFilename,
+    path.resolve(destPath, indexFilename + "x"),
+  );
 })();
 
 // GPP leaves the files littered with these; they alter parsing by flipping sections to HTML context
 // remove them BEFORE parsing to avoid issues
-function stripEmptyComments(file)
-{
-  file.contents = file.contents.toString().replace(/<!--\s*-->/g, '');
+function stripEmptyComments(file) {
+  file.contents = file.contents.toString().replace(/<!--\s*-->/g, "");
```

Comment:
[suggest] I'm not a fan of functions that mutate incoming params. If possible, I would refactor so there are no side-effects to calling a function.

Reply:
Reasonable; I've adjusted the other two BDR-ecosystem scripts in this regard as well. (at some point need to adapt Jake's scripts to a unifiedjs pipeline and remove all the redundancy here)